nixos/modules/misc/wordlist.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+# Processes the source in the option
+# Returns either the path passed in if it's a string, or a new path to the
+# wordlist constructed from the various paths and strings.
+processSource = name: source:
+  if builtins.typeOf source == "string"
+  then
+    source
+  else # typeOf source is list of tagged data
+    builtins.toString (makeFileFromList name source);
+
+# Concatenate a list of sources (paths or strings) into a single file and
+# return the path to that file.
+makeFileFromList = name: sources:
+  pkgs.writeText name (lib.foldl (acc: source: acc + processOne source) "" sources);
+
+# Processes a single source from inside the list
+# Returns the contents of the wordlist if passed in a path, otherwise joins the
+# list of strings with newlines
+processOne = source:
+  if source ? file
+  then
+    builtins.readFile source.file
```

Comment:
IFD?

Reply:
import from derivation. if you supply a path from a package as `source.file` that package has to be built during the evaluation phase, which is a single-threaded process and thus slows down a lot. for that reason (and others) it's disallowed in some installations, so it's best to avoid doing things that require nix to read files from derivation outputs during eval.